src/runner/api/itemRunner.js

```diff
@@ -0,0 +1,467 @@
+                        if (_.isFunction(pendingTask)) {
+                            pendingTask.call(self);
+                        }
+                    });
+                    flow.render.pending = [];
+                }
+
+                /**
+                 * The item is rendered
+                 * @event itemRunner#render
+                 */
+                self.trigger('render');
+
+                /**
+                 * The item is ready.
+                 * Alias of {@link itemRunner#render}
+                 * @event itemRunner#ready
+                 */
+                self.trigger('ready');
+            };
+
+            options = _.defaults(options || {}, { state: {} });
+
+            //check elt
+            if (!(elt instanceof HTMLElement) && !(elt instanceof $)) {
+                return self.trigger(
+                    'error',
+                    'A valid HTMLElement (or a jquery element) at least is required to render the item'
+                );
+            }
+
+            //set item state to restore item state after rendering if the provider enables it
+            if (options.state) {
+                this.setState(options.state);
+            }
+
+            if (flow.init.done === false) {
+                flow.init.pending.push(function() {
+                    this.render(elt, options);
+                });
+            } else {
+                //we keep a reference to the container
+                if (elt instanceof $) {
+                    this.container = elt.get(0);
+                } else {
+                    this.container = elt;
+                }
+
+                //the state will be applied only when the rendering is made
+
+                if (_.isFunction(provider.render)) {
+                    /**
+                     * Calls the provider's render
+                     * @callback RendertItemProvider
+                     * @param {HTMLElement} elt - the element to render inside
+                     * @param {Function} done - call once the render is done
+                     * @param {Object} [options] - the array of options that the item runner provider may supports
+                     * @param {Object} [options.state] - pass initial item state to method render() in case the item runner provider require initial state to render
+                     */
+                    provider.render.call(this, this.container, renderDone, options);
+                } else {
+                    renderDone();
+                }
+            }
+
+            return this;
+        },
+
+        /**
+         * Clear the running item.
+         * @returns {itemRunner}
+         *
+         * @fires itemRunner#clear
+         */
+        clear: function() {
+            var self = this;
+
+            /**
+             * Call back when clear is done
+             */
+            var clearDone = function clearDone() {
+                //events = {};
```

Comment:
I guess this was the purpose for `events`. But it seems to be abandoned code now.